graphql_compiler/compiler/filters.py

```diff
@@ -829,6 +829,11 @@ def _get_filter_op_name_and_values(directive):
     if 'op_name' not in args:
         raise AssertionError(u'op_name not found in filter directive arguments!'
                              u'Validation should have caught this: {}'.format(directive))
+    if args['op_name'].value.value in UNARY_FILTERS and 'value' not in args:
+        args['value'] = Argument(Name('value'), ListValue([], loc=args['op_name'].loc))
```

Comment:
This seems rather circuitous, and potentially dangerous. It's not clear whether `args` contains references to objects provided by the input parameter, and so it's unclear whether we are mutating function inputs or not.

Why are we taking this risk instead of refactoring the code to not require this kind of mutation?

Reply:
Let me know what you think of the new code